onnxruntime/core/platform/linux/device_discovery.cc

```diff
@@ -2,11 +2,13 @@
 // Licensed under the MIT License.
 
 #include "core/platform/device_discovery.h"
+#include "core/framework/ortdevice.h"
```

Comment:
usually, we want to stick to this dependency ordering:

https://github.com/microsoft/onnxruntime/blob/5ef8ce2103e862a1c14166ed7a406f88b990927e/cmake/onnxruntime.cmake#L238-L254

so onnxruntime_common (which contains core/platform/linux/device_discovery.cc) should not depend on onnxruntime_framework (core/framework/ortdevice.h).

we can move the vendor id enum and helper function to be a separate header, in onnxruntime_common perhaps, or just move the helper function to this file for now.

Reply:
I've put the helper to the linux file. I will try to generalize in a follow-up.